completion.go

```diff
@@ -0,0 +1,145 @@
+package kong
+
+import (
+	"reflect"
+)
+
+//CompletionOptions options for shell completion.
+type CompletionOptions struct {
+	// the function to run for shell completions. If nil, no completions will be run. default: nil
+	Completer Completer
+
+	// installer/uninstaller for shell completions.  If nil, there will be no install or uninstall flags
+	CompletionInstaller CompletionInstaller
+
+	// flag name for installing completions to a shell.  default: --install-shell-completion
+	InstallFlagName string
+
+	// help text for --install-shell-completion. default: "Install shell completion."
+	InstallFlagHelp string
+
+	// flag name for uninstalling completions from a shell.  default: --uninstall-shell-completion
+	UninstallFlagName string
+
+	// help text for --uninstall-shell-completion. default: "Uninstall shell completion."
+	UninstallFlagHelp string
+
+	// whether to hide --install-shell-completion and --uninstall-shell-completion from help.  default: false
+	HideFlags bool
```

Comment:
I'd prefer the flags to just be separate types (as you have below with `uninstallCompletionsValue`), then the users can decide how they want to use them as they would any other flag. It will also reduce the complexity of this config struct.

Reply:
I think you are saying that we can get rid of `HideFlags` all the `*Name` and `*Help` fields above and replace them with exported types that people can put in their own cli structs with whatever names, help messages or other attributes they see fit.  If so, that sounds good to me.